src/devices/Pcd8544/Pcd8544.cs

```diff
@@ -0,0 +1,487 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Gpio;
+using System.Device.Spi;
+using System.Device.Pwm;
+using System.Threading;
+using System.Drawing;
+using Iot.Device.Display.Pcd8544Enums;
+
+namespace Iot.Device.Display
+{
+    /// <summary>
+    /// PCD8544 - 48 × 84 pixels matrix LCD
+    /// </summary>
+    public class Pcd8544 : IDisposable
```

Comment:
I think it should be possible to implement `ICharacterLcd` (from the CharacterLCD package). This will allow using the LcdCharacterEncoding capabilities, which provides help to work with different cultures.

Reply:
I was hesitating as the screen does not embedded all the capabilities. But I will look at it.

Reply:
There is no blinking cursor for example in the screen or no real concept of cursor.
And there is no concept of font at all, reason why I just embed the original one.
So that's the reason I only picked few of the functions. But I'll be happy to understand how to make this better integrated if you have an idea.
This device is actually closer to Ssd13xx than the ones in CharacterLcd

Reply:
I think we should not try to do that as part of this PR. I'd rather start thinking on display abstraction again and then think of some sort of solution to make all displays implement character lcd (i.e. create some class which adds the functionality to any display)

Reply:
I'm fine with delaying that for later, but actually that interface is exactly intended to be implemented by various displays to provide support for common functionality (such as text wrapping, culture dependent character translations etc).

Reply:
And what about for screens aligning with the API of Azure GUIX: https://docs.microsoft.com/fr-fr/azure/rtos/guix/about-guix?
So having something that is inspired from that?
Couple of comments on the naming in there are related as well to this choice.

Reply:
my suggestion would be to go with ImageSharp and then build on top of that (although haven't read about GUIX that much just ImageSharp is recommended drawing library now) but let's discuss this outside of this PR

Reply:
By the way, this will be an issue: https://docs.sixlabors.com/articles/imagesharp/pixelformats.html#is-it-possible-to-store-a-pixel-on-a-single-bit-for-monochrome-images
If we can't support 1 bit for monochrome screens, almost none of those b&w screens will be supported. While having ImageSharp for high level design may work, we will still always have to rework on the generated buffer.
Also, they don't provide any interface or anything like this for screens

Reply:
@pgrawehr btw if I implement fully the ICharacterLcd , then there are function which will throw a not implemented, is that acceptable?

Reply:
@Ellerbach I suppose it would be better to just ignore unsupported calls, especially if there's no side effect to be expected for the caller if it doesn't work. I.e. if your display doesn't support enabling/disabling the backlight, then the getter would just always return true and the setter wouldn't do anything. A cursor (although not blinking) could be implemented by printing a _ letter.

Reply:
We could also be discussing about removing these properties from the interface or move them to a derived one, to make the thing more generic.

Reply:
@pgrawehr, let me give a try. And let see how it can look like.
@krwq I fully migrated it to ImageSharp. Was quite straight forward. See this commit

Reply:
@pgrawehr, I gave it a try and I'm bock because now I4m using ImageSharp, the ICharacterLcd is not and it's still using System.Drawing. See [for example](https://github.com/dotnet/iot/blob/ca828c3d28d34976a8e07d24357e029e66b87905/src/devices/CharacterLcd/ICharacterLcd.cs#L38)
For the rest I can manage. That's more than doable. For the cursor, it will be a bit weird as it's not blinking, underlining should be manageable. 
For custom Characters, here is the question: technically, you can really create almost as many as you want, it's just about having a list of byte[] in the binding and that's it. Now because the Characters can be any format but so far are aligned with the Nokia font embedded, they are byte[] of 5.
The doc says byte[] of 8. [See](https://github.com/dotnet/iot/blob/ca828c3d28d34976a8e07d24357e029e66b87905/src/devices/CharacterLcd/ICharacterLcd.cs#L91)
So I can then just return 0, meaning no special character can be allowed. Also, then when displaying them, I don't know how to properly handle this.
So all up, if you remove the custom character map from the interface and move to ImageSharp, then I'm sure that can be actually a pretty nice interface for screens as well. At least for a basic usage, that will perfectly work!

Reply:
@Ellerbach The Type `System.Drawing.Size` is in `System.Drawing.Primitives.dll`, not in `System.Drawing.Common.dll`, which seems to be part of the framework. Are you sure this should also be removed? Not a big deal, but still this is a different assembly.

The custom characters for the LCD displays currently use row-oriented byte[8] characters. The characters are 5x8 pixels, each byte represents a row (only the lower 5 bits used). The documentation on the interface is actually wrong, as only the implementation defines the expected size, not the interface. So if your implementation takes byte[5], that's basically fine. 

Since the font size in the end is also 5x8, you could (automatically) transpose those bitmasks to the same format. Then you could basically declare the `NumberOfCustomCharacters` as 256 and implement the font as an encoding. This would even (since the font sizes are equal) allow the use of the existing font maps for the custom characters I have added already. I can understand if that last part sounds a bit complicated, but I can have a look at it myself if you want.

Reply:
Regarding `System.Drawing`,from the issue #1403 and thid: https://docs.microsoft.com/en-us/dotnet/api/system.drawing?view=dotnet-plat-ext-5.0#remarks, I guess we should really move out fully.
I know it's not much but if we move, we have to move all. @krwq any thoughts?

Regarding the cataracter set, ok, got it, so that can be done. I can then have this implemented. The new characters will replace the default ones if one is added with the same number. This will allow fully customize fonts.
Got it for the font orientation. I can even detect if it's byte[5] or byte[8], assuming if it's byte[8], it's then 5 rows and transforms it.

Thought on this?

Reply:
Sounds good. Just be aware that the custom characters so far do not have fixed numbers. Their place in the character map is determined by the culture and the fixed font.

Reply:
> Their place in the character map is determined by the culture and the fixed font.

Can you give me an example so it's more clear to me?

Reply:
You'll find relevant samples in `LcdConsole.Samples.cs` (if you've done everything right<sup>tm</sup>, that example should work against your interface implementation with a minor change only). 

Basically the workflow is as follows, given an instance of `ICharacterLcd`:

```csharp
ICharacterLcd lcd=new Pcd8544(...)
LcdConsole console = new LcdConsole(lcd, "NoFontRom", false); // Use a descriptive string for your "empty" character rom
console.Write(...); // *
LcdCharacterEncoding encoding = LcdConsole.CreateEncoding(CultureInfo.CreateSpecificCulture("de-CH"), "NoFontRom", '?', 256); // Provide Rom name again, and number of available custom characters. This knows which letters the given language requires (i.e. in this case äöüÄÖÜ) and internally maps these to the free slots. So for instance, "ä" might become character value 0.
console.LoadEncoding(encoding); // Load the encoding. This also loads all required custom characters.
console.Write(...) // Now this uses the address translation from the encoding to know that the string "äää" shall actually print 3 times the letter with character value 0.

// * That would now do nothing visible, since there's no default font. - case to consider separately, because using the display without an explicit character set will lead to unexpected behavior, also with the current implementation. Currently, it defaults to a simple cast of char to byte, but I consider changing this to an exception instead.
```

While writing the above, I found that it seems the `LcdCharacterEncoding` class is missing (at least) a few `virtual` declarations to make the scenario work with a fully customized font. I think we should have a call on the details.

Reply:
@pgrawehr I implemented the ICharacterLcd and all the cursor part plus some needed functions, had to do some refactoring/transformation to fully and properly support cursors, now working all good. Moved the CharacterLcd namespace to ImageSharp.
Still some work on the font side now. Trying to understand the best way to handle all this, based as well on our discussion. That's next on the list!

Reply:
@krwq and @pgrawehr I would go for merge for this PR as it implements all the basics for the CharacterLcd and a certain level of compatibility. Now to support fonts and few other advanced scenarios, that would require more changes in both classes (we've discussed with @pgrawehr some of the implication). That's then a work I can do later on as I won't really have time to make this soon.
By merging this one, it can be on the 1.4 and we have a decent support so the screen can be used like a LCD as well.

Reply:
@krwq and @pgrawehr I added 2 special fonts 5x7 and 5x8 containing a lot of unicode characters to cover almost all scenarios. Those had been autogenerated.
I added as well the support for those specific characters for the screen. Example updated.
I had to update the ILcdCharacter for this additional support. I don't have any small screen to test on it. So a deep test should be done before merge. Rest seems to work on my Nokia screen.